File: yy/common/LibHNMarket/platform/android/HNPay/src/com/hn/pay/base/ProductInfo.java
package com.hn.pay.base;

import org.json.JSONException;
import org.json.JSONObject;


public class ProductInfo {
    // ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½
	public long				number = 0;
    // ï¿½ï¿½Æ·ï¿½Û¸ï¿½
	public double			price = 0;
    // ï¿½ï¿½Æ·ID
	public String           identifier = null;
	// Ö§ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ sdk
	public int				payType = -1;
	
	public String           xmlFile = null;
	
	// ¶©µ¥ºÅ
    public String           orderID = null;
	
	public ProductInfo()
    {    
    }
	
    public ProductInfo(long number, double price, String identifier, int payType, String orderID)
    {
        // ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½
    	this.number = number;
        // ï¿½ï¿½Æ·ï¿½Û¸ï¿½
    	this.price = price;
        // ï¿½ï¿½Æ·ID
    	this.identifier = identifier;       
    	// Ö§ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ sdk
    	this.payType = payType;
    	// ¶©µ¥ºÅ
    	this.orderID = orderID;
    }

    public String serialize()
    {
    	return "";
    }
    
    public void deserialize(String data)
    {
    	try {
			JSONObject json = new JSONObject(data);
			this.number = json.getInt("number");
			this.price = json.getDouble("price");
			this.identifier = json.getString("identifier");
			this.payType = json.getInt("payType");
			this.xmlFile = json.getString("xmlFile");
			this.orderID = json.getString("orderID");
		} catch (JSONException e) {
			e.printStackTrace();
		}
    }
}

File: yy/common/LibHNLobby/HNLobby/Shop/ProductInfo.cpp
#include "HNLobby/Shop/ProductInfo.h"
#include "json/rapidjson.h"
#include "json/document.h"
#include "json/stringbuffer.h"
#include "json/writer.h"

namespace HN 
{
	std::string PRODUCT_INFO::serialize()
	{
		rapidjson::Document doc;
		doc.SetObject();
		rapidjson::Document::AllocatorType& allocator = doc.GetAllocator();
		rapidjson::Value object(rapidjson::kObjectType);
		object.AddMember("identifier", rapidjson::StringRef(identifier.c_str()), allocator);
		object.AddMember("number", number, allocator);
		object.AddMember("price", price, allocator);
		object.AddMember("payType", payType, allocator);
		object.AddMember("xmlFile", rapidjson::StringRef(xmlFile.c_str()), allocator);
		object.AddMember("orderID", rapidjson::StringRef(orderID.c_str()), allocator);
		rapidjson::StringBuffer buffer;
		rapidjson::Writer<rapidjson::StringBuffer> write(buffer);
		object.Accept(write);
		return buffer.GetString();
	}

	void PRODUCT_INFO::deserialize(std::string data)
	{
		rapidjson::Document doc;
		doc.Parse<rapidjson::kParseDefaultFlags>(data.c_str());
		if (!doc.HasParseError())
		{
			if (doc.IsObject() && doc.HasMember("status"))
			{
				identifier = doc["identifier"].GetString();
				number = doc["number"].GetInt64();
				price = doc["price"].GetDouble();
				payType = doc["payType"].GetInt();
				xmlFile = doc["xmlFile"].GetString();
				orderID = doc["orderID"].GetString();
			}
		}
	}
}                                                  